src/main/kotlin/no/nav/syfo/service/SenderFacade.kt

```diff
@@ -20,26 +18,48 @@ import no.nav.syfo.kafka.producers.dittsykefravaer.DittSykefravaerMeldingKafkaPr
 import no.nav.syfo.kafka.producers.dittsykefravaer.domain.DittSykefravaerVarsel
 
 class SenderFacade(
-    val dineSykmeldteHendelseKafkaProducer: DineSykmeldteHendelseKafkaProducer,
-    val dittSykefravaerMeldingKafkaProducer: DittSykefravaerMeldingKafkaProducer,
-    val brukernotifikasjonerService: BrukernotifikasjonerService,
-    val arbeidsgiverNotifikasjonService: ArbeidsgiverNotifikasjonService,
-    val fysiskBrevUtsendingService: FysiskBrevUtsendingService,
-    val database: DatabaseInterface
+    private val dineSykmeldteHendelseKafkaProducer: DineSykmeldteHendelseKafkaProducer,
+    private val dittSykefravaerMeldingKafkaProducer: DittSykefravaerMeldingKafkaProducer,
+    private val brukernotifikasjonerService: BrukernotifikasjonerService,
+    private val arbeidsgiverNotifikasjonService: ArbeidsgiverNotifikasjonService,
+    private val fysiskBrevUtsendingService: FysiskBrevUtsendingService,
+    val database: DatabaseInterface,
 ) {
     fun sendTilDineSykmeldte(
         varselHendelse: NarmesteLederHendelse,
         varsel: DineSykmeldteVarsel,
     ) {
-        dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)
+        try {
+            dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)
+        } catch (e: Exception) {
+            log.error("Error while sending varsel to DINE_SYKMELDTE: ${e.message}", e)
+            lagreIkkeUtsendtNarmesteLederVarsel(
+                kanal = DINE_SYKMELDTE,
+                varselHendelse = varselHendelse,
+                eksternReferanse = varsel.id.toString(),
+                feilmelding = e.message,
+            )
+        }
         lagreUtsendtNarmesteLederVarsel(DINE_SYKMELDTE, varselHendelse, varsel.id.toString())
```

Comment:
Skal `lagreUtsendtNarmesteLederVarsel(DINE_SYKMELDTE, varselHendelse, varsel.id.toString())` inn i `try`-blokken, slik at varsel ikke lagres som utsendt dersom det oppstår `Exception`? Evt . kaste en exception slik du gjør her: https://github.com/navikt/esyfovarsel/pull/290/files#diff-e090391d68c4c788dfe4191b65b8c8b185aa33c99aeb706238b0be11ca877b87R115?

Reply:
Try/catchen og rethrowing av Exception du peker på er for å fange feil som skjer under utsending til eksterne tjenester, da lagres det varsel i den nye tabellen og rethrowes ny exception. Etter  ny exception er rethrown avsluttes code execution og `lagreUtsendtNarmesteLederVarsel(DINE_SYKMELDTE, varselHendelse, varsel.id.toString())` ikke blir kalt om jeg forstår riktig(ingen `finally` her). Mener du å fange feil som kan skje når varsel lagres i database?

Reply:
@Vitenok Men her blir vel ikke exception rethrowet? Så hvis det oppstår en Exception i `dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)` (linje 33), så blir både `lagreIkkeUtsendtNarmesteLederVarsel(...) og `lagreUtsendtNarmesteLederVarsel(...)` på hhv.
linje 36 og 43 kalt. Er det ikke meningen at varsel kun skal lagres som utsendt hvis det ikke feiler
under utsending til eksterne?

Reply:
stemmer. beklager. Fikser nå

Reply:
👍

Reply:
Da er det gjort @torchr89